### ISSUE TITLE
Cant change locale on 404 page

### ISSUE THREAD

**Issue by Alex289** [vercel/next.js#32549]
### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

16.11.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

Vercel

### Describe the Bug

I have a project with internationalized routing and locally everything works as expected but when I deploy my application to Vercel I cant change the locale on the 404 page anymore.

### Expected Behavior

Change the locale even on the 404 page

### To Reproduce

Create a nextjs application and add i18n with two locales. Then add a custom 404 page returning a next/link tag changing just the locale.
This is my link tag:
```ts
<Link
  href={router.pathname}
  scroll={false}
  shallow={true}
  locale={locale === 'de' ? 'en' : 'de'}>
    {locale === 'de' ? 'EN' : 'DE'}
</Link>
```

Deploy this application to Vercel, go to the 404 page and try to change the locale.

This is the error messages I get (The first 404 is expected since its the not found page):
![Capture](https://user-images.githubusercontent.com/38790209/146252197-37fb0f3f-be70-4b31-8ad9-5f9774433734.PNG)

The repo I deployed ans has this issue can be found [here](https://github.com/Alex289/Portfolio)

**Comment by Alex289** [comment]
Anyone there?

**Comment by Alex289** [comment]
My issue got solved with the release of next 12.0.10 I guess.
